pyam/core.py

```diff
@@ -782,10 +788,60 @@ def check_aggregate(self, variable, components=None, exclude_on_fail=False,
 
             return IamDataFrame(diff, variable=variable).timeseries()
 
+    def weighted_average_region(self, variable, weight, region='World',
+                                subregions=None, append=False):
+        """Compute weighted average of a timeseries over multiple regions
+
+        Parameters
+        ----------
+        variable: str
+            variable for which the aggregate should be computed
+         weight: str
+            weighting variable to be used for calculating weighted average
+        region: str, default 'World'
+            dimension
+        subregions: list of str
+            list of subregions, defaults to all regions other than `region`
+        append: bool, default False
+            append the aggregate timeseries to `data` and return None,
+            else return aggregate timeseries
+        """
+        # default subregions to all regions other than `region`
+        subregions = subregions or self._all_other_regions(region, variable)
+
+        subregion_df = self.filter(region=subregions)
+        cols = ['region', 'variable']
+
+        weightvar_df = subregion_df.filter(variable=weight).data
+
+        if weightvar_df.empty:
+            msg = "cannot aggregate because no data found "\
+                "or weight_var='{}'"
+            logger().error(msg.format(weight))
+            return
+        var_df = subregion_df.filter(variable=variable).data
+        # _data = _aggregate(var_df, cols,
+        #                    method='w.avg', weightvar_df=weightvar_df)
+        cols = [c for c in list(var_df.columns) if c not in ['value'] + cols]
```

Comment:
it's partly personal preference, but I think that the PR https://github.com/danielhuppmann/pyam/pull/12 is a far more pythonic and **readable** implementation